analysis/tests/logic_test.py

```diff
@@ -0,0 +1,43 @@
+import inspect
```

Comment:
1. How should I run this test? I tried `python tests/logic_test.py`, but even if I remove `[150, 1000]` from insane data it returns 0 exit code.
2. I think we want to run this test on CI. @balsoft @mkaito can help with that. AFAIU `analysis/default.nix` should have some `analysis-test` that we'll add to `checks` in the root `flake.nix` and then add `analysis test` step to `pipeline.yml`.

Reply:
To run all tests you run `pytest`. I checked, if I delete `[150, 1000]`, this test fails. Tbh, I plan to write more tests, that's why I added other files, just didn't have time yet and we agreed to have a dummy PR by today.

Reply:
Yes, if I run `pytest` I get a nicely looking output and the test fails without that point. So we just need to run `pytest` on CI then.

> Tbh, I plan to write more tests, that's why I added other files, just didn't have time yet and we agreed to have a dummy PR by today.

Ok, got it.